### PR TITLE
Support ORDER BY and BYTES in data anonymizer tool

### PR DESCRIPTION
**Pull request by siddharthteotia**
(1) ORDER BY in query generator
(2) BYTES type support for both global dictionary and
non global dictionary columns

This is a follow-up to https://github.com/apache/incubator-pinot/pull/4905 as we address few small limitations of the tool and make it ready for enhancing an internal test framework

### CODE REVIEW

File: pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java [hunk 1]
```diff
@@ -530,68 +544,107 @@ private void generateDerivedValuesForGlobalDictionary(OrigAndDerivedValueHolder
...
 
-  private String[] generateDerivedStringValues(OrigAndDerivedValueHolder valueHolder) {
+  private String[] generateDerivedStringValuesForGD(OrigAndDerivedValueHolder valueHolder) {
     int cardinality = valueHolder._index;
     String[] values = new String[cardinality];
     for (int i = 0; i < cardinality; i++) {
-      int origValLength = ((String)valueHolder._origValues[i]).length();
+      String val = (String)valueHolder._origValues[i];
+      int origValLength;
+      if (val == null || val.equals("") || val.equals(" ") || val.equals("null")) {
```
Comment (mcvsubbu) [inline pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java]: `if (val == null || val.equals("null)) {
  // choose "null" as the GD value
}`

File: pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java [hunk 2]
```diff
@@ -530,68 +544,107 @@ private void generateDerivedValuesForGlobalDictionary(OrigAndDerivedValueHolder
...
   }
 
+  private ByteArray[] generateDerivedByteValuesForGD(OrigAndDerivedValueHolder valueHolder) {
+    int cardinality = valueHolder._index;
+    ByteArray[] values = new ByteArray[cardinality];
+    Random random = new Random();
+    for (int i = 0; i < cardinality; i++) {
+      ByteArray byteArray = (ByteArray)valueHolder._origValues[i];
+      int origValLength;
+      if (byteArray == null || byteArray.length() == 0) {
```
Comment (mcvsubbu) [inline pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java]: same as before for Strings, leave it at empty byte array

File: pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java [hunk 3]
```diff
@@ -1070,6 +1132,25 @@ private String rewriteGroupBy(GroupByAstNode groupByAstNode) {
...
+    private String rewriteOrderBy(OrderByAstNode orderByAstNode) {
+      StringBuilder orderBy = new StringBuilder();
+      orderBy.append("ORDER BY ");
+      List<? extends AstNode> children = orderByAstNode.getChildren();
+      int count = 0;
+      for (AstNode orderByChild : children) {
+        if (count > 0) {
+          orderBy.append(", ");
+        }
+        Preconditions.checkState(orderByChild instanceof OrderByExpressionAstNode);
```
Comment (mcvsubbu) [inline pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java]: we dont need the pre-condition. it is the same as throwing the class cast exception in the next line, right? (unless you are providing some meaningful exception string here)

File: pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java (hunk 1 above)
Comment (siddharthteotia) [reply pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java]: done

File: pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java (hunk 2 above)
Comment (siddharthteotia) [reply pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java]: done

File: pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java (hunk 3 above)
Comment (siddharthteotia) [reply pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java]: yes. removed

**Comment by siddharthteotia** [comment]
@mcvsubbu , addressed the comments

**Pull request merged**